xonsh/completer.py

```diff
@@ -200,19 +187,25 @@ def complete(self, prefix, line, begidx, endidx, ctx=None):
             # anything goes.
             rtn = self.cmd_complete(prefix)
         elif cmd in self.bash_complete_funcs:
-            rtn = set()
-            for s in self.bash_complete(prefix, line, begidx, endidx):
-                if os.path.isdir(s.rstrip()):
-                    s = s.rstrip() + slash
-                rtn.add(s)
-            if len(rtn) == 0:
-                rtn = self.path_complete(prefix, path_str_start, path_str_end)
+            # bash completions
+            rtn = self.bash_complete(prefix, line, begidx, endidx)
+            rtn |= self.path_complete(prefix, path_str_start, path_str_end)
+            return self._filter_repeats(rtn), lprefix
         elif prefix.startswith('${') or prefix.startswith('@('):
             # python mode explicitly
-            rtn = set()
+            return _python_mode_completions(self, prefix, prefixlow, startswither)
         elif prefix.startswith('-'):
             return sorted(self._man_completer.option_complete(prefix, cmd)), lprefix
         elif cmd not in ctx:
+            ltoks = line.split()
+            if len(ltoks) > 2:
+                if ltoks[0] == 'from' and ltoks[2] == 'import':
```

Comment:
Ah, you learn something new every day.  I had also intended to wrap this in a try/except in case the import fails.